File: single_player/tutorial/ruby/onboarding.rb
STDOUT.sync = true # DO NOT REMOVE
# CodinGame planet is being attacked by slimy insectoid aliens.


# game loop
loop do
    $enemy1 = gets.chomp # name of enemy 1
    $dist1 = gets.to_i # distance to enemy 1
    $enemy2 = gets.chomp # name of enemy 2
    $dist2 = gets.to_i # distance to enemy 2
    
    # Write an action using puts
    # To debug: STDERR.puts "Debug messages..."
    
    puts ($dist2 > $dist1) ? $enemy1 : $enemy2
end

File: single_player/tutorial/java/onboarding.java
import java.util.Scanner;

class Player {
    public static void main(String args[]) {
        Scanner in = new Scanner(System.in);

        while (true) {
            String enemy1 = in.next();
            int dist1 = in.nextInt();
            String enemy2 = in.next();
            int dist2 = in.nextInt();
            
            System.out.println((dist1 < dist2) ? enemy1 : enemy2);
        }
    }
}